core/rawdb/bor_receipt.go

```diff
@@ -0,0 +1,130 @@
+package rawdb
+
+import (
+	"github.com/maticnetwork/bor/common"
+	"github.com/maticnetwork/bor/core/types"
+	"github.com/maticnetwork/bor/ethdb"
+	"github.com/maticnetwork/bor/log"
+	"github.com/maticnetwork/bor/rlp"
+)
+
+var (
+	borReceiptPrefix = []byte("bor-receipt-") // borReceiptPrefix + number + block hash -> bor block receipt
+
+	// freezerBorReceiptTable indicates the name of the freezer bor receipts table.
+	freezerBorReceiptTable = "bor-receipts"
+)
+
+// borReceiptKey = borReceiptPrefix + num (uint64 big endian) + hash
+func borReceiptKey(number uint64, hash common.Hash) []byte {
+	return append(append(borReceiptPrefix, encodeBlockNumber(number)...), hash.Bytes()...)
+}
+
+// HasBorReceipt verifies the existence of all block receipt belonging
+// to a block.
+func HasBorReceipt(db ethdb.Reader, hash common.Hash, number uint64) bool {
+	if has, err := db.Ancient(freezerHashTable, number); err == nil && common.BytesToHash(has) == hash {
+		return true
+	}
+
+	if has, err := db.Has(borReceiptKey(number, hash)); !has || err != nil {
+		return false
+	}
+
+	return true
+}
+
+// ReadBorReceiptRLP retrieves all the transaction receipts belonging to a block in RLP encoding.
```

Comment:
`all the transaction receipts` or `block receipts`?

Reply:
It's a typo. I will fix it.